### ISSUE TITLE
Regarding Issue of Date Range Filter not giving appropriate results.

### ISSUE THREAD

**Issue by Jaski96** [adobe/asset-share-commons#806]
**Describe the bug**
We are having issues in getting all search results for date range filter. In this filter, we have two input fields for Start Date and End Date. When we select start date and then end date and click apply button to get search results, then it's not giving us results specifically for assets on end date.

**Environment**
* AEM Version: 2022.7.8085.20220725T140323Z
* Asset Share Commons Version: 2.2.0-11
* Issue encountered in both Author and Publish.

**Expected behavior**
It should give us all search results including ones on End Date.

**Screenshots**
We used sample asset having license expiration date set as 2022-9-2. As shown below, We added dates in Start Date and End Date and then click on apply button to apply filter.
![image](https://user-images.githubusercontent.com/105772836/188150830-72f49330-ade7-41d6-aa64-e06b4415c8d9.png)

After that page would give message "Your search did not match any assets." Here we should get asset in search result.
![image](https://user-images.githubusercontent.com/105772836/188152019-b7cb4db8-321e-4603-8ecc-bd551ab54167.png)

**Comment by davidjgonzalez** [comment]
@Jaski96 can you show a screenshot of the component configuration? thanks!

**Comment by Jaski96** [comment]
![image](https://user-images.githubusercontent.com/105772836/188257240-05e310ce-f2ef-44c4-be3b-e0c93a606225.png)
Assets on end date as shown in above comment. We have set license expiration date as 2022-9-2 for End Date input field but asset is not retrieved in search result.

**Comment by davidjgonzalez** [comment]
@Jaski96 the date range seems to work for me -- are you sure that the property is being saved as a Date field? and indexed as such as well?

**Comment by Jaski96** [comment]
@davidjgonzalez Let me explain in two scenarios.
**Scenario-1** I used one asset with expiration date set as 2022-09-22. So, now If I use date range filter by setting date for start  date and end date as 20 Sep and 22 Sep respectively as shown in below screenshot and click apply button for searching.
![image](https://user-images.githubusercontent.com/105772836/191674207-d0524b9a-75ad-4677-91aa-3bd1bd291b30.png)

Having following authoring component configuration.
![image](https://user-images.githubusercontent.com/105772836/191673735-d0742562-01a1-401f-9259-957f9d242495.png)
![image](https://user-images.githubusercontent.com/105772836/191673877-d220c1a4-1ecd-40ea-82be-e41ae34adb57.png)

In this case, I don't get asset in end date but I get asset from start date.

**Scenario-2** I used one asset with expiration date set as 2022-09-22. So, now If I use date range filter by setting date for start  date and end date as 20 Sep and 22 Sep respectively as shown in below screenshot.
